### PR TITLE
Base64 encode the scope value

### LINKED ISSUE #50: bkt should make scope "filename-safe"
My very first use of `bkt` was to cache the output of a `kubectl` command. I want the results scoped to a particular cluster, so I did this:

```
export BKT_SCOPE=$(kubectl config view --minify -o json | jq .clusters[0].cluster.server)
```

This ends up with `BKT_SCOPE` having a value like `https://api.cluster.example.com:6443`, which ultimately causes `bkt` to fail with:

```
bkt: Cache write failed: No such file or directory (os error 2)
```

And this is because it's trying to use the scope literally as part of a pathname:

```
1142219 rename("/tmp/bkt-0.7-cache/keys/tmp-symlink.TysE1Bh1WR7I7TDH", "/tmp/bkt-0.7-cache/keys/\"https://api.cluster.example.com:6443\".4EBF7E540DA5C7C7") = -1 ENOENT (No such file or directory)
```

Either...

(a) The documentation should make it clear that scope value is used verbatim in a pathname and must not contain any "path sensitive" characters (and `bkt` should probably throw an error if it detects things like `/` or `\` in the scope value), or

(b) `bkt` should sanitize the path (e.g., to only ascii alphanumerics or something), so that the above might result in the path `.../keys/https___api_cluster_example_com_6443...`.

(c) or maybe just base64-encode the scope value.

### PR DESCRIPTION
**Pull request by larsks**
bkt permits setting a cache scope (provided via the `BKT_SCOPE`
environment variable or the `--scope` command line option) to prevent
collisions between unrelated command invocations. Previously, the scope
value was used verbatim in a directory path, causing errors if the scope
contained path-sensitive characters such as `/`.

This commit modifies bkt to base64 encode the scope value. Given a scope
value like `https://example.com/`, that means we now end up with a path
like:

    /tmp/bkt-0.7-cache/keys/aHR0cHM6Ly9leGFtcGxlLmNvbS8.C3823B193F1E0C78

Instead of the invalid:

    /tmp/bkt-0.7-cache/keys/https://www.example.com/.C3823B193F1E0C78

Fixes #50

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -627,7 +630,7 @@ impl Cache {
 
     fn scoped(mut self, scope: String) -> Self {
         assert!(self.scope.is_none());
-        self.scope = Some(scope);
+        self.scope = Some(general_purpose::STANDARD_NO_PAD.encode(scope));
```
Comment (dimo414) [inline src/lib.rs:633]: Let's do the encoding [where the path is constructed](https://github.com/dimo414/bkt/blob/b01ecb612b2aa37e84b79a0340d23c1d165bd655/src/lib.rs#L667), instead of here

**Review by dimo414 — COMMENTED** [review]
Thank you! Welcome to Rust 😄 

I'd appreciate if you could add test coverage as well; you should be able to duplicate [this test](https://github.com/dimo414/bkt/blob/b01ecb612b2aa37e84b79a0340d23c1d165bd655/src/lib.rs#L920) with a scope that breaks prior to this PR.

**Comment by larsks** [comment]
Thanks for the feedback. I'll try to take care of these sometime this week.

**Comment by larsks** [comment]
@dimo414 just following up to see if this needs any more work. Thanks!

**Comment by dimo414** [comment]
Sorry I had missed that you'd updated the PR (you can click the "re-request review" button to notify the reviewer - GitHub's review flow is pretty confusing :confused:). Will take a look as soon as I can.

File: src/lib.rs [hunk 2]
```diff
@@ -931,6 +934,24 @@ mod cache_tests {
         assert_eq!(present_scoped.unwrap().0, val_b);
     }
 
+    #[test]
+    fn scoped_fails_without_base64() {
```
Comment (dimo414) [inline src/lib.rs:938]:
```suggestion
    fn scopes_support_special_chars() {
```

**Comment by dimo414** [comment]
Thanks again! Will plan to cut a new release with this soon.
